docs/ide/editorconfig-code-style-settings-reference.md

```diff
@@ -0,0 +1,896 @@
+
+C#:
+```C#
+var list = new List<int>(); 
+list.Add(1);
+list.Add(2);
+list.Add(3);
+```
+Visual Basic: 
+```VB
+Dim list = new List(Of Integer)
+list.Add(1)
+list.Add(2)
+list.Add(3)
+```
+
+#### Example:
+```editorconfig
+# CSharp and VisualBasic code style settings:
+[*.cs, *.vb]
+dotnet_style_collection_initializer = true:suggestion
+```
+
+### Prefer explicit tuple names
+#### Option Name:
+```
+dotnet_style_explicit_tuple_names
+```
+
+#### Applicable Languages: C# and Visual Basic
+
+#### Description:
+**true:** Prefer tuple names to ItemX properties.
+
+C#:
+```C#
+(string name, int age) customer = GetCustomer(); var name = customer.name;
+```
+Visual Basic: 
+```VB
+Dim customer As (name As String, age As Integer) = GetCustomer()
+Dim name = customer.name;
+```
+
+**false:** Prefer ItemX properties to tuple names.
+C#:
+```C#
+(string name, int age) customer = GetCustomer(); var name = customer.Item1;
+```
+Visual Basic: 
+```VB
+Dim customer As (name As String, age As Integer) = GetCustomer()
+Dim name = customer.Item1
+```
+
+#### Example:
+```editorconfig
+# CSharp and VisualBasic code style settings:
+[*.cs, *.vb]
+dotnet_style_explicit_tuple_names = true:suggestion
+``` 
+
+### Prefer coalescing expressions in "null" checking
+#### Option Name:
+```
+dotnet_style_coalesce_expression
+```
+
+#### Applicable Languages: C# and Visual Basic
+
+#### Description:
+**true:** Prefer null coalescing expression to ternary operator checking.
+
+C#:
+```C#
+var v = x ?? y
+```
+Visual Basic: 
+```VB
+Dim v = If(x, y)
+```
+
+**false:** Prefer ternary operator checking to null coalescing expression.
+
+C#:
+```C#
+var v = x != null ? x : y; // or
+var v = x == null ? y : x;
+```
+Visual Basic: 
+```VB
+Dim v = If(x Is Nothing, y, x) ' or
+Dim v = IF(x IsNot Nothing, x, y)
+```
+
+#### Example:
+```editorconfig
+# CSharp and VisualBasic code style settings:
+[*.cs, *.vb]
+dotnet_style_coalesce_expression = true:suggestion
+``` 
+
+### Prefer null propagation in "null" checking
+#### Option Name:
+```
+dotnet_style_null_propagation
+```
+
+#### Applicable Languages: C# and Visual Basic
+
+#### Description: 
+**true:** Prefer to use null-conditional operator where possible.
+
+C#:
+```C#
+var v = o?.ToString();
+```
+Visual Basic: 
+```VB
+Dim v = o?.ToString()
+```
+
+**false:** Prefer to use ternary null checking where possible.
+
+C#:
+```C#
+var v = o == null ? null : o.ToString(); // or
+var v = o != null ? o.String() : null;
+```
+Visual Basic: 
+```VB
+Dim v = If(o Is Nothing, Nothing, o.ToString()) ' or
+Dim v = If(o IsNot Nothing, o.ToString(), Nothing)
+```
+
+#### Example:
+```editorconfig
+# CSharp and VisualBasic code style settings:
+[*.cs, *.vb]
+dotnet_style_null_propagation = true:suggestion
+``` 
+
+# CSharp Code Style Settings
+## Prefer "var"
+### Prefer "var" for built-in types
+#### Option Name:
+```
+csharp_style_var_for_built_in_types
+```
+
+#### Applicable Languages: C#

```

Comment:
Not sure that where you have "applicable languages" makes sense? Can we add it as a subtitle under the the big headings?